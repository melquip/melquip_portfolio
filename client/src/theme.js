export default {
  preload: [
    'https://fonts.googleapis.com/css?family=Odibee+Sans|Roboto+Mono&display=swap',
  ],
  colors: {
    black: 'rgba(17, 17, 17, 1)',
    white: '#fff',
    lightgray: '#eee',
    orange: 'rgba(239, 148, 108, 1)',
    lightorange: 'rgba(239, 148, 108, .5)',
    purple: 'rgba(75, 78, 109, 1)',
    blue: 'rgba(38, 84, 124, 1)',
    midblue: 'rgba(38, 84, 124, .8)',
    lightblue: 'rgba(38, 84, 124, .25)',
    red: 'rgba(239, 71, 111, 1)',
  },
  fonts: {
    primary: '"Roboto Mono", monotype, sans-serif',
    secondary: '"Odibee Sans", "Roboto Mono", cursive, sans-serif'
  },
  mediaTiny: 'screen and (min-width: 360px)',
  mediaMobile: 'screen and (min-width: 480px)',
  mediaTablet: 'screen and (min-width: 720px)',
  mediaDesktop: 'screen and (min-width: 1024px)',
  mediaLarger: 'screen and (min-width: 1280px)',
}