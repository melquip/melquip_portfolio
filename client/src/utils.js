export const loadState = (keys) => {
	try {
		let serializedState = null;
		let err = false;
		serializedState = keys.reduce((serialSt, key) => {
			serialSt[key] = localStorage.getItem('melquip_' + key);
			if(serialSt[key] !== null) {
				serialSt[key] = JSON.parse(serialSt[key]);
			} else {
				err = true;
			}
			return serialSt;
		}, {})
		return serializedState === null || err ? undefined : serializedState;
	} catch (err) {
		console.error('loadState:', err);
		return undefined;
	}
}

export const saveState = (state, keys) => {
	try {
		let serializedState = null;
		if (keys.length > 0) {
			keys.forEach(key => {
				serializedState = JSON.stringify(state[key]);
				localStorage.setItem('melquip_' + key, serializedState);
			});
		} else {
			serializedState = JSON.stringify(state);
			localStorage.setItem('melquip_state', serializedState);
		}
	} catch (err) {
		throw new Error('Not able to set local storage: ', err);
	}
}