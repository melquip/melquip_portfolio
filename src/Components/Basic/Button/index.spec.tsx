// #region Global Imports
import * as React from "react";
// #endregion Global Imports

// #region Local Imports
import { render, fireEvent } from "@Test/utils";
import { theme } from "@Definitions/Styled/theme";
import { Button } from "./index";
// #endregion Local Imports

describe("Basic Components", () => {
    describe("Button", () => {
        it("should have the passed class", () => {
            const { container } = render(<Button className="active" />);

            expect(container.firstChild).toHaveClass("active");
        });

        it("should render with children prop", () => {
            const { getByText } = render(<Button>hey</Button>);

            expect(getByText("hey")).toBeTruthy();
        });

        it("should increment number on click", () => {
            let number = 1;
            const { getByText } = render(
                <Button
                    onClick={() => {
                        number += 1;
                    }}
                >
                    Button
                </Button>
            );

            fireEvent.click(getByText("Button"), new MouseEvent("click"));

            expect(number).toBe(2);
        });

        it("should match snapshot", () => {
            const { container } = render(<Button>Test</Button>);

            expect(container).toMatchSnapshot();
        });

        it("should be disabled", () => {
            const { container } = render(<Button disabled>Test</Button>);

            expect(container.firstChild).toHaveStyleRule(
                "cursor",
                "not-allowed"
            );
        });

        it("should be enabled", () => {
            const { container } = render(
                <Button disabled={false}>Test</Button>
            );

            expect(container.firstChild).toHaveStyleRule("cursor", "pointer");
        });

        it("should have primary color", () => {
            const { container } = render(<Button>Test</Button>);

            expect(container.firstChild).toHaveStyleRule(
                "color",
                theme.colors.primary
            );
        });
    });
});
