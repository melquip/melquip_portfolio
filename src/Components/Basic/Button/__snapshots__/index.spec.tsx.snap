// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Components Button should match snapshot 1`] = `
.c0 {
  cursor: pointer;
  color: #2c3e50;
}

<div>
  <div
    class="c0"
  >
    Test
  </div>
</div>
`;
